bootstrap: fall back to the default upstream URL

mathstuf opened the pull request:
Fixes #352.

---
@JohnWestlund @tgamblin

tgamblin commented:
@mathstuf: any chance you could make this also use the currently checked out branch to bootstrap, instead of master?

mathstuf commented:
@tgamblin Yeah, it forced the `master` branch. Changes since last time:

  - ability to choose which remote to use;
  - select `develop` by default;
  - names the branch in the bootstrapped directory.

mathstuf commented:
Anything blocking this still?
